src/app/dashboard/dashboard.js

```diff
@@ -1,34 +1,33 @@
 import { Component } from '@angular/core';
-import { ROUTER_DIRECTIVES } from '@angular/router';
+import { ROUTER_DIRECTIVES, ActivatedRoute } from '@angular/router';
 
 import style from './style.scss';
 import template from './dashboard.html';
 import { SensorWidget } from './sensor-widget';
 import { SensorExecutorWidget } from './sensor-executor-widget';
 import { SensorStatusWidget } from './sensor-status-widget';
 import SensorWidgetService from './shared/sensor-widget.service';
-import DashboardService from './dashboard.service';
 
 @Component({
     selector: 'sm-dashboard',
     template,
     styles: [style],
     directives: [ROUTER_DIRECTIVES, SensorWidget, SensorExecutorWidget, SensorStatusWidget],
-    providers: [DashboardService, SensorWidgetService]
+    providers: [SensorWidgetService]
 })
 export class Dashboard {
-    constructor(dashboardService: DashboardService, sensorWidgetService: SensorWidgetService) {
-        this.dashboardService = dashboardService;
+    constructor(sensorWidgetService: SensorWidgetService,
+                route: ActivatedRoute) {
         this.sensorWidgetService = sensorWidgetService;
+        this.route = route;
         this.widgets = [];
     }
 
     ngOnInit() {
-        this.dashboardService
-            .getWidgets()
-            .subscribe(({devices}) => {
-                this.widgets = devices;
-            });
+        this.route.data.subscribe(({devices}) => {
```

Comment:
How about use approach with static params ```this.route.snapshot.params``` instead of using dynamic route params?
Because we are using ```devices``` only one time